corehq/motech/repeaters/models.py

```diff
@@ -195,32 +241,63 @@ def all_ready(self):
                 .filter(repeat_records_ready_to_send))
 
 
-class SQLRepeater(models.Model):
-    domain = models.CharField(max_length=126)
-    repeater_id = models.CharField(max_length=36)
+class OptionValue(property):
+
+    NOT_SET = object()
+
+    def __init__(self, default=NOT_SET, choices=None):
+        self.default = default
+        self.choices = choices
+
+    def __set_name__(self, owner, name):
+        self.name = name
+
+    def __get__(self, obj, objtype=None):
+        if obj is None:
+            return self
+        if self.name in obj.options:
+            return obj.options[self.name]
+        if self.default is self.NOT_SET:
+            raise AttributeError(self.name)
+        value = self.default() if callable(self.default) else self.default
+        obj.options[self.name] = value
+        return value
+
+    def __set__(self, obj, value):
+        if self.choices and value not in self.choices:
+            raise ValueError(f"{value!r} not in {self.choices!r}")
+        obj.options[self.name] = value
+
+
+class SQLRepeater(SyncSQLToCouchMixin, RepeaterSuperProxy):
```

Comment:
Haven't been paying much attention to this work lately, but ideally this will get renamed back to `Repeater` so the interim name won't matter. Is that going to be possible here @AmitPhulera ?

Reply:
> Haven't been paying much attention to this work lately, but ideally this will get renamed back to Repeater so the interim name won't matter. Is that going to be possible here?

Yes, Jenny, I believe it would be renamed to Repeater after we are done with the migrations. 

> Is there a reason why we made it a prefix here? Would it make sense to switch it to a suffix to be more consistent with other SQL model classes?

I guess I followed with Norman had initially started and did not think more about the naming.  Yes, it would be possible to rename the classes but as Jenny pointed out that it would be in the codebase for a brief period of time. So should we be going with the rename?